### PR TITLE
Use shopping list card in panel

### PR DESCRIPTION
**Pull request by iantrich**
## Proposed change

Was looking at remaining Polymer code and thought it is probably time to retire the shopping list panel, especially with in the incoming enhancements to the card in https://github.com/home-assistant/frontend/pull/7296

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [x] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Review by zsarnett — APPROVED** [review]
Does this require a backend change as well? I know for the Calendar Panel it setup in the back to show in the sidebar

**Comment by iantrich** [comment]
@zsarnett I don't know that answer actually. Perhaps @bramkragten knows, otherwise I'll give a look sometime later

**Comment by zsarnett** [comment]
@iantrich 

https://github.com/home-assistant/core/blob/2a795c039742c31a72323e382a98e89e0d1fbfa9/homeassistant/components/shopping_list/__init__.py#L110-L111

**Comment by bramkragten** [comment]
So the idea is that a `shopping-list` card in panel mode would be the same? I would no longer be automatically setup though and thus is harder to find?

We could also make this panel use the card? Or are there other reasons to remove this panel?

**Comment by balloob** [comment]
I agree with Bram that this makes the shopping list impossible to find. This is not a good user experience. Since we can also disable panels now, I don't see a reason to drop the panel just because it's still Polymer.

**Comment by iantrich** [comment]
The card being more feature Rich was the motivation to drop it, I found by looking for remaining Polymer elements. I like the idea of using the card in the panel.

**Comment by balloob** [comment]
Just render the Shopping list panel as a Lovelace dashboard with 1 card 😆

**Comment by iantrich** [comment]
image

So one this I forgot about is the microphone which gives the ability to add/delete via voice. Not sure how much that is used, but if we want that functionality back it should be put into the card IMO

File: src/panels/shopping-list/ha-panel-shopping-list.ts [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+        #columns {
+          display: flex;
+          flex-direction: row;
+          justify-content: center;
+          margin-left: 4px;
+          margin-right: 4px;
+        }
+        .column {
+          flex: 1 0 0;
+          max-width: 500px;
```
Comment (bramkragten) [inline src/panels/shopping-list/ha-panel-shopping-list.ts:95]: Should we add a min-width and a columns/column? Why not just will width?

image

Comment (iantrich) [reply src/panels/shopping-list/ha-panel-shopping-list.ts:95]: I had that initially and it just seemed way too big. Old panel had it centered as I have it now and thought it looked a bit better. 🤷‍♂️I'm really indifferent though.

**Comment by zsarnett** [comment]
I think we should just add the microphone to the panel like we do in Lovelace

**Comment by iantrich** [comment]
> I think we should just add the microphone to the panel like we do in Lovelace

for some reason I was thinking it was a shopping list specialized microphone thing, I never use it 😅. Yeah will add it.

File: src/panels/shopping-list/ha-panel-shopping-list.ts (hunk 1 above)
Comment (balloob) [reply src/panels/shopping-list/ha-panel-shopping-list.ts:118]: I agree that full width is not what we want. It means the buttons at the end are too far away from the actual content.

File: build-scripts/gulp/translations.js [hunk 2]
```diff
@@ -43,7 +43,6 @@ const TRANSLATION_FRAGMENTS = [
   "logbook",
   "mailbox",
   "profile",
-  "shopping-list",
```
Comment (balloob) [inline build-scripts/gulp/translations.js:46]: Should this be put back.

Comment (balloob) [reply build-scripts/gulp/translations.js:46]: Or maybe even better, we should auto-generate this based on the `src/panels` folders and then add the pages.

Comment (bramkragten) [reply build-scripts/gulp/translations.js:46]: There are no translations in the panel 🤷‍♂️

File: src/panels/shopping-list/ha-panel-shopping-list.ts [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+            <ha-menu-button
+              .hass=${this.hass}
+              .narrow=${this.narrow}
+            ></ha-menu-button>
+            <div main-title>${this.hass.localize("panel.shopping_list")}</div>
+            ${this._conversation(this.hass.config.components)
+              ? html`
+                  <mwc-icon-button
+                    .label=${this.hass!.localize(
+                      "ui.panel.lovelace.menu.start_conversation"
```
Comment (bramkragten) [inline src/panels/shopping-list/ha-panel-shopping-list.ts:69]: We can't use translation tags from other panels, as they will not be available until that panel is loaded.

Comment (iantrich) [reply src/panels/shopping-list/ha-panel-shopping-list.ts:69]: Will that effect the translations for the shopping list card itself as it does use lovelace translations in it?

**Review by bramkragten — APPROVED** [review]

**Pull request merged**